### ISSUE TITLE
Header propagation does not work when RestClient is called from a @ServerEndpoint context

### ISSUE THREAD

**Issue by philcolol** [quarkusio/quarkus#12125]
We are using 'org.eclipse.microprofile.rest.client.propagateHeaders' property together with @RegisterClientHeaders annotation to propogate Authorization header to RestClients. It works when rest client called from Rest endpoints but fails with 401  when called from Webcosket endpoints. Is there some other configuration or well-known way to fix this?

**To Reproduce**
Consider RestClient for some external service:
```
@Path("/externalService")
@RegisterRestClient
@RegisterClientHeaders
public interface DataService {

    @GET
    @Path("/method")
    @Produces("application/json")
    Set<String> getData();
}
```

A websocket endpoint that calls it:
```
@ServerEndpoint("/endpoint")         
@ApplicationScoped
public class WsEndpoint {

    @Inject
    ManagedExecutor executor;

    @Inject
    ThreadContext threadContext;

    @Inject
    @RestClient
    DataService dataService;

    @OnMessage
    public void onMessage(String message) {
        executor.submit(threadContext.contextualRunnable(() -> {
            session.getAsyncRemote().sendObject(dataService.getData()))));
        }
    }

}
```

And a request
```
curl \
    --include \
    --no-buffer \
    -H "Connection: Upgrade" \
    -H "Upgrade: websocket" \
    -H "Sec-WebSocket-Key: wPu/9zumjcR93EmDazjRFw==" \
    -H "Sec-WebSocket-Version: 13" \
    -H "Authorization: Bearer ey..." \
    http://127.0.0.1:8086/endpoint
```

**Configuration**
```properties
# Add your application.properties here, if applicable.
org.eclipse.microprofile.rest.client.propagateHeaders=Authorization
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
Microsoft Windows [Version 10.0.18363.1016]
 - Output of `java -version`: 
openjdk version "1.8.0_212-release"
OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4)
OpenJDK 64-Bit Server VM (build 25.212-b4, mixed mode)
 - Quarkus version or git rev: 
1.3.2.Final, also reproduced on 1.7.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\...\apache-maven-3.6.3
Java version: 1.8.0_192, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk1.8.0_192\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

**Comment by sberyozkin** [comment]
@philcolol Is this header visible to the `@ServerEndpoint` ? I thought the HTTP headers were not available in the web socket channel, CC @stuartwdouglas

**Comment by stuartwdouglas** [comment]
This definitely won't work at the moment, but in theory it should be possible to make it work.

**Comment by kenfinnigan** [comment]
Agreed. The MP REST Client spec only details propagating headers from an incoming JAX-RS request.

**Comment by dondragon2** [comment]
Has this been resolved? any workarounds?
